### ISSUE TITLE
[introduction/quickstart/preparing-the-cluster]: [Further detail for hugepage allocation]

### ISSUE THREAD

**Issue by AntonOfTheWoods** [openebs/mayastor#513]
**Are you reporting an issue with existing content?**

A warning should be given to users that 512x2MB hugepages may not be enough, as it appears that other software will use hugepages if they are available but don't *need* them. This can be surprising for users where hugepages are not used, as they naturally assume that adding 512 will satisfy the 512 that are required by mayastor. When other processes "steal" hugepages, this means that upon reboot, there will not be enough hugepages for the MSN, and MSN resources won't be created by the daemonset nodes.

https://mayastor.gitbook.io/introduction/quickstart/preparing-the-cluster

**Are you proposing new content, or a change to the existing documentation layout or structure?**

Add a warning that users should make sure to verify that the hugepages are not being used by other, opportunistic processes and, if possible, to provision *more* than 512 pages, or at least to check that after a reboot there continues to be *at least* 512 hugepages that are still available in `HugePages_Free`.

Basically the examples should NOT be with:
```
vm.nr_hugepages = 512
```
They should be something like 
```
vm.nr_hugepages = 1024
```
Or possibly 
```
vm.nr_hugepages = 512 # PLUS whatever is taken after a reboot, PLUS a margin for other applications
```
Another option could be to have the daemonset pods error out, though I have no idea whether there is a usecase for these if the node doesn't have hugepages.

**Comment by jkryl** [comment]
Right. IMO we should keep 512 in our examples and document that on systems with other applications using hugepages one needs to allocate more. How much depends on the application requirements.

In addition to that we could perhaps check # of free hugepages in init container of mayastor pod to be at least 512. Not sure if that would make sense. What do you think @tiagolobocastro ?

**Comment by tiagolobocastro** [comment]
We could do, but what would we do if there are not at least 512 available?
What is the current behaviour, does the mayastor container actually run or kubelet won't even schedule mayastor on the node if the node is not able to fulfil the pages we set in the spec?

**Comment by AntonOfTheWoods** [comment]
I was getting nothing, my understanding is that it is the pods themselves that create the resources when they launch, and I was not getting any pods or resources. As a total beginner with mayastor, this was quite confusing.

**Comment by jkryl** [comment]
@tiagolobocastro : indeed. My proposal with init container does not make sense as there is nothing we can do in the init container if there are not enough hugepages. Waiting for hugepages to appear is not an option.

Much better would be to check the requirement in mayastor itself, log error message and exit cleanly if < 512. That would be trivial to do.

**Comment by gila** [comment]
K8s is keeping track of the resources claimed in POD spec. So other
applications when deployed via k8s, should not be a problem as the
scheduler would account for the pages already claimed.

are you referring to applications on the node not started by k8s?

**Comment by AntonOfTheWoods** [comment]
I didn't have anything else installed on the nodes. One consequence I found was that Bitnami's postgresql-ha helm chart's postgres instances we not launching properly after I enabled hugepages..
